`openapi-validator` - missing types for `request` and `superagent`

```ts
node_modules/openapi-validator/dist/classes/RequestPromiseResponse.d.ts:1:40 - error TS2307: Cannot find module 'request' or its corresponding type declarations.

1 import type { Request, Response } from 'request';
                                         ~~~~~~~~~

node_modules/openapi-validator/dist/classes/SuperAgentResponse.d.ts:1:50 - error TS2307: Cannot find module 'superagent' or its corresponding type declarations.

1 import type { Response, SuperAgentRequest } from 'superagent';
                                                   ~~~~~~~~~~~~

```
`@types/request` and `@types/superagent` should probably be listed as `dependencies` instead of `devDependencies` as they are required downstream if using you're using typescript and don't want to enable `skipLibCheck`

Thanks for the explanation and SO link! I've managed to recreate this. I agree `request` and `superagent` should be prod deps. I found that `axios` is missing, so we should add that too. I'll:
1. merge your PR
2. raise and merge another one for `axios`
3. publish this to npm in a new minor version, check it compiles without errors

### Investigation

(Are you importing `openapi-validator` directly or either `jest-openapi` or `chai-openapi-response-validator`?) 

If I import only `openapi-validator` into a project, I get your 2 errors (`request` and `superagent`), and an `axios` error:
```ts
node_modules/openapi-validator/dist/classes/AxiosResponse.d.ts:1:57 - error TS2307: Cannot find module 'axios' or its corresponding type declarations.

1 import type { AxiosResponse as AxiosResponseType } from 'axios';
                                                          ~~~~~~~
```

I'm not sure why you don't get the `axios` error. `openapi-validator` [uses it](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc475535417bf97c67864599bcdcdcc781/packages/openapi-validator/lib/classes/AxiosResponse.ts#L1) but doesn't list it in [its `package.json`](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc47/packages/openapi-validator/package.json#L49). The tests pass because this is a monorepo, and the jest and chai plugins list [`axios` in their `package.json`s](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc475535417bf97c67864599bcdcdcc781/packages/chai-openapi-response-validator/package.json?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L48). This is pretty fragile, so however we fix `request` and `superagent`, we should do the same for `axios`.

If I import `chai-openapi-response-validator` instead of `openapi-validator`, then I get the 3 errors above, and 2 chai errors:
```ts
node_modules/chai-openapi-response-validator/dist/index.d.ts:1:23 - error TS2688: Cannot find type definition file for 'chai'.

1 /// <reference types="chai" />
                        ~~~~

node_modules/chai-openapi-response-validator/dist/index.d.ts:19:78 - error TS2694: Namespace 'global.Chai' has no exported member 'ChaiPlugin'.

19 export default function (filepathOrObject: string | OpenAPISpecObject): Chai.ChaiPlugin;
```
Similar to above, `chai-openapi-response-validator` [lists `chai` as a devDep](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc47/packages/chai-openapi-response-validator/package.json#L49), not a prod dep. However, users will normally have chai installed, so won't hit this error. We could list chai as a peer deps, but it doesn't matter because no one can use `chai-openapi-response-validator` without `chai`. I think I'll leave this as is.

So back to `request`, `superagent`, and `axios`. Should they be prod deps or peer deps? If they are prod deps, they should just work. [In theory](https://stackoverflow.com/a/34645112/9864252) there could be a type conflict if someone uses two versions of a package together. But we don't - if a user does `expect(axiosResponse).toSatisfyApiSpec`, that `axiosResponse` is treated as `any`, so it can be cast to [our internal `axios` type](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc475535417bf97c67864599bcdcdcc781/packages/chai-openapi-response-validator/lib/assertions/satisfyApiSpec.ts#L19) without conflict. So making them prod deps will work.

Setting them as peer deps wouldn't solve the problem. Users would still get the errors you found until they install the 3 packages manually (as you're doing) or skipLibCheck. npm / yarn will display a peer dep warning, but it's easy to miss.

So yes, let's make `request`, `superagent`, and `axios` prod deps of `openapi-validator`.


### Notes
If I remove `@types/jest` and `jest`, `jest-openapi` doesn't have compile errors . Not sure why.

`openapi-validator` uses `openapi-types` but lists it as a _dev_ dep not a prod dep. We don't get a compilation error because the chai and jest plugins have a prod dep `openapi-response-validator` which includes `openapi-types`. That's a bit fragile. However, if we list `openapi-types` as a prod dep, we _could_ get type conflicts, because the [argument to `chaiResponseValidator(filepathOrObject)`](https://github.com/openapi-library/OpenAPIValidators/blob/8a388bdc475535417bf97c67864599bcdcdcc781/packages/chai-openapi-response-validator/lib/index.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L24) expects a string or an object satisfying the OpenAPI type from our version of `openapi-types`. So if a user provides an object satisfying the type of a different version of `openapi-types`, there could be a compilation error. That might be good actually. But I don't have much experience with this so I'd prefer to leave it until someone asks for it and there's a clear benefit

--- a/packages/openapi-validator/package.json
+++ b/packages/openapi-validator/package.json
@@ -37,6 +37,8 @@
     "dist"
   ],
   "dependencies": {
+    "@types/request": "^2.48.7",
+    "@types/superagent": "^4.1.12",
     "combos": "^0.2.0",
     "fs-extra": "^9.0.0",
     "js-yaml": "^4.0.0",
@@ -48,8 +50,6 @@
   "devDependencies": {
     "@types/fs-extra": "^9.0.12",
     "@types/js-yaml": "^4.0.3",
-    "@types/request": "^2.48.7",
-    "@types/superagent": "^4.1.12",
     "@types/typeof": "^1.0.0",
     "openapi-types": "^9.2.0"
   }

